erts/emulator/beam/erl_bif_binary.c

```diff
@@ -2457,6 +2459,53 @@ static int cleanup_copy_bin_state(Binary *bp)
     return 1;
 }
 
+static BIF_RETTYPE binary_secure_compare(Process *p, Eterm bin1, Eterm bin2)
```

Comment:
This implementation differs from the original implementation in #2749 in that we do not short circuit if lengths do not match. Instead we always iterate based on the size of the largest binary which feels like the right thing to do. I think if you want to short circuit on differing lengths that is particular to a caller's case and thus should be handled by the caller.